### PR TITLE
Added golangci-lint workflow action

### PR DESCRIPTION
**Pull request by Raviv-S**
Added golangci-lint workflow action for github and deleted an old linter workflow: golang-security.yml

Signed-off-by: Raviv Schaffer <raviv.schaffer@ibm.com>

### CODE REVIEW

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v2
+        with:
+          version: latest
+          args: --new-from-rev=HEAD~
+          level: warning # GitHub Status Check won't become failure with this level.
```
Comment (roytman) [inline .github/workflows/golangci-lint.yml]: why do you use the warning level? We do want that the action fails if it doesn't pass

Comment (Raviv-S) [reply .github/workflows/golangci-lint.yml]: I've copied this section from the file you've uploaded on your PR

File: hack/make-rules/verify.mk [hunk 2]
```diff
@@ -38,48 +38,6 @@ CODE_MAINT += tidy
...
-
-.PHONY: lint-fix
-lint-fix: $(TOOLBIN)/golangci-lint
-	$(TOOLBIN)/golangci-lint run --fix ${GOLINT_LINTERS} ./...
-
-.PHONY: lint-todo
-lint-todo: $(TOOLBIN)/golangci-lint
-	$(TOOLBIN)/golangci-lint run --enable=godox ${GOLINT_LINTERS} ./...
-
-.PHONY: misspell
```
Comment (roytman) [inline hack/make-rules/verify.mk:75]: why did you remove it?

File: hack/make-rules/verify.mk [hunk 3]
```diff
@@ -38,48 +38,6 @@ CODE_MAINT += tidy
...
-misspell: $(TOOLBIN)/misspell
-	$(TOOLBIN)/misspell --error ./**
-
-.PHONY: misspell-fix
```
Comment (roytman) [inline hack/make-rules/verify.mk]: and this

File: .github/workflows/golangci-lint.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v2
+        with:
+          version: latest
+          args: --new-from-rev=HEAD~
```
Comment (roytman) [inline .github/workflows/golangci-lint.yml]: in the small repositories, where you fixed the code, you don't need `--new-from-rev=HEAD~`, it can check entire repository

**Review by roytman — APPROVED** [review]
